### PR TITLE
Change option that enables/disables tier 0 JIT and expose another related option

### PR DESCRIPTION
**Pull request by kouvel**
Depends on https://github.com/dotnet/coreclr/pull/23597

- Tier 0 JIT is being referred to as quick JIT. Renamed option that was added by https://github.com/dotnet/sdk/pull/2925 to TieredCompilationQuickJit, which indicates whether tiered compilation may use quick JIT in its startup tier. Default will be false.
- May be enabled in .csproj under `PropertyGroup` element with `<TieredCompilationQuickJit>true</TieredCompilationQuickJit>`
- Or in `<app>.runtimeconfig.json` with `"System.Runtime.TieredCompilation.QuickJit": true`

### CODE REVIEW

**Comment by kouvel** [comment]
CC @noahfalk @nguerrera

I'd like to get this into preview 4, please let me know if I should port this to the 3.0.1xx branch. Also do changes that go into 3.0.1xx get mirrored automatically to master?

**Comment by kouvel** [comment]
CC @jkotas @richlander

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets
Comment (jkotas) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]: I am not sure whether it make sense to have `QuickJitForLoops` exposed in SDK first class. I look at it as one of those config switches for experts only.

**Comment by nguerrera** [comment]
> I'd like to get this into preview 4, please let me know if I should port this to the 3.0.1xx branch

We haven't snapped for preview 4 from master yet, so this is fine.

**Review by nguerrera — APPROVED** [review]

**Comment by kouvel** [comment]
Flagged with `[WIP]` in title due to dependency. Thanks @nguerrera, it may be a bit longer before this can be merged due to the dependency. I'll check with you again before merging to make sure it's going into the right branch for preview 4.

**Comment by nguerrera** [comment]
This needs to go in by Friday for Preview 4. After that, you will need M2 level approval in addition to changing the target branch.

**Comment by kouvel** [comment]
Thanks for the heads-up @nguerrera, I'll try to get this in before then

**Review by noahfalk — APPROVED** [review]

**Comment by kouvel** [comment]
Thanks @noahfalk. @nguerrera, I think this is ready to be merged but I don't have permissions to do so, would you be able to merge it?

**Comment by wli3** [comment]
@nguerrera  FYI i merged it

**Comment by nguerrera** [comment]
Thank, @wli3

**Pull request merged**
